### ISSUE TITLE
Limited size for environment variables

### ISSUE THREAD

**Issue by fmalheiro** [cloudfoundry/cloud_controller_ng#1148]
## Issue

Need to add multiple (somewhat large) environment variables to an application.

## Context

While doing successive `cf set-env`s we hit a hard limit to what can be added to the environment variables of an application. 

Looking at the [code](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/models/runtime/app_model.rb#L30), all environment variables are being serialized to one column of the database. Consequently, the size of the sum of all the environment variables is limited to that of the column.

## Steps to Reproduce

By doing something like,

`cf set-env AppName Key1 LargeValue1`
`cf set-env AppName Key2 LargeValue2`
...
`cf set-env AppName KeyN LargeValueN`

Eventually, the limit will be hit.

## Expected result

Be able to set all of the environment variables.

## Current result

When the limit is hit, adding any new environment variable will output something like:

```
Setting env variable 'ThisIsOneOfManyEnvironmentVariableKeys' to 'ThisIsOneOfManyEnvironmentVariableValues' for app AmazingApp in org test / space Default as admin...
FAILED
Server error, status code: 500, error code: 10011, message: Database error
```

**Comment by cwlbraa** [comment]
Hi, @fmalheiro . I can see a strong argument for having a better error message for these failures than "Database error", but I'm afraid it is far from straightforward for the cc to meet your expected result. Here's an incomplete list of the difficulties:

- multi tenant environments still need limits on the size of this data
- the current limit is implicit on the max size of our ORM's `String, text: true` column, which happens to be very different for different db backends (65,535 chars for MySQL but 1GB+ for postgres)
- both key and value need to remain encrypted at rest to maintain our security guarantees
- this is a pretty large dataset in large prod environments, and changing its schema would likely incur api downtime. the encryption and encoding aspect of that migration likely make it cpu-time intensive.
- encoding+encryption of this data means that size at rest != size on the wire
- imposing new configuration for size limits gets awkward fast. It's easy to state, impose, and enforce a limit on "size on the wire" in a new environment, but in long lived environments, it's not easy to figure out how big the data at rest would've been on the wire, nor is it easy to "grandfather" in apps that violate new limits

Beyond giving an error more descriptive than 'database error', in the near term it seems unlikely product will prioritize improving this. Would `Server error, status code: 400, error code: XXXX, message: Environment variable size exceeds database limits` have made this less painful to encounter? The initial error seems very user-unfriendly.

**Comment by charleshansen** [comment]
Agreed it is not a good idea to remove the limit on environment variables. A better error message would be nice, but I'm hesitant to prioritize it due to the technical weirdness of catching this type of db error (see the fix in #1233). Let us know if the error message would have solved a lot of the pain.

**Comment by aashah** [comment]
Looks like this has been addressed in #1555, closing.
